common/djangoapps/student/models.py

```diff
@@ -1244,6 +1244,26 @@ def get_enrollment(cls, user, course_key, select_related=None):
         except cls.DoesNotExist:
             return None
 
+    @classmethod
+    def get_program_enrollment(cls, user, course_id):
+        """
+        Return the ProgramEnrollment associated with the CourseEnrollment specified
+        by the user and course_id.
+        Return None if there is no ProgramEnrollment.
+
+        Arguments:
+            user (User): the user for whom we want the program enrollment
+            coure_id (CourseKey): the id of the course the user has a course enrollment in
+
+        Returns:
+            ProgramEnrollment object or None
+        """
+        try:
+            course_enrollment = cls.objects.get(user=user, course_id=course_id)
+            return course_enrollment.programcourseenrollment.program_enrollment
+        except (CourseEnrollment.DoesNotExist, ObjectDoesNotExist):
```

Comment:
`CourseEnrollment.DoesNotExist` is a subclass of `ObjectDoesNotExist`, so this is redundant